### ISSUE TITLE
`torch.compile` (inductor) — `interpolate(mode='nearest')` resamples a spatial dimension whose size is unchanged, disagreeing with eager

### ISSUE THREAD

**Issue by tinywisdom** [pytorch/pytorch#198201]
### 🐛 Describe the bug

`F.interpolate(x, scale_factor=..., mode='nearest')` disagrees between eager and
`torch.compile` (inductor) when a `scale_factor` is non-integer but the resulting output size
equals the input size for that dimension. Eager treats such an **unchanged spatial dimension as
identity** (source index = destination index, no resampling); the compiled lowering still applies
the user-provided non-unit scale factor and resamples that dimension, selecting different source
pixels. Same output shape, different values, no error.

## Minimal repro

```python
import torch
import torch.nn.functional as F

# H: 3 -> floor(3*1.3)=3 (UNCHANGED) with a non-unit scale;  W: 4 -> floor(4*1.5)=6 (changed)
x = torch.arange(12, dtype=torch.float32).reshape(1, 1, 3, 4)   # CPU, float32

def fn(z):
    return F.interpolate(z, scale_factor=(1.3, 1.5), mode='nearest')

e = fn(x)
c = torch.compile(fn, backend="inductor")(x)

# height source-row mapping (first column of each output row):
print([int(e[0, 0, i, 0]) // 4 for i in range(3)])   # eager:    [0, 1, 2]  (unchanged dim -> identity)
print([int(c[0, 0, i, 0]) // 4 for i in range(3)])   # compiled: [0, 0, 1]  (resampled with scale 1.3)
print(torch.equal(e, c))                              # False
```

## Expected vs actual

- **Expected**: compiled matches eager — a dimension whose output size equals its input size is
  passed through unchanged (identity), so the height rows map [0, 1, 2].
- **Actual**: inductor resamples the unchanged height dimension using the raw `scale_factor=1.3`,
  mapping rows [0, 0, 1] — different pixels, silently.

## Scope (observed, CPU and CUDA identical)

| dim | input→output | scale | eager vs compiled |
|---|---|---|---|
| output size **==** input size | 3→3 | 1.3 | **DIFF** (eager identity, compiled resamples) |
| output size **==** input size | 4→4 | 1.2 | **DIFF** |
| output size changes | 3→6 | 2.0 | same |
| output size changes (shrink) | 3→2 | 0.7 | same |

Only the "unchanged size + non-unit scale" case diverges; dimensions that actually change size
agree. Reproduces on both CPU and CUDA (eager's equal-size shortcut is device-independent).

### Versions

```
PyTorch version: 2.15.0a0+gite67e43b
Is debug build: True
CUDA used to build PyTorch: 12.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: (Ubuntu 12.3.0-1ubuntu1~22.04.3) 12.3.0
Clang version: 17.0.6
CMake version: version 4.0.0
Libc version: glibc-2.35

Python version: 3.12.13 | packaged by conda-forge (64-bit runtime)
Python platform: Linux-6.8.0-138-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 12.6.20
GPU 0: NVIDIA RTX A6000
Nvidia driver version: 580.82.07
cuDNN version: 9.3.0

[pip3] numpy==2.3.5
[pip3] torch==2.15.0a0+gite67e43b
[pip3] triton==3.7.1
```

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo
